src/drawline.c

```diff
@@ -838,7 +838,13 @@ wlv_screen_line(win_T *wp, winlinevars_T *wlv, int negative_width)
 	{
 	    ScreenLines[off] = '<';
 	    if (enc_utf8)
+	    {
 		ScreenLinesUC[off] = 0;
+		if (i + 1 + skip < wp->w_width && ScreenLines[off + 1] == 0)
+		    // When the first half of a double-width character is
+		    // overwritten, change the second half to a space.
+		    ScreenLines[off + 1] = ' ';
```

Comment:
For 'listchars' "precedes", `MB_FILLER_CHAR` is used, but using it here leads to `>>><`, which looks a bit strange.